core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.java

```diff
@@ -56,31 +58,28 @@
 import static org.kiwix.kiwixmobile.core.wifi_hotspot.HotspotService.ACTION_STOP_SERVER;
 
 public class ZimHostActivity extends BaseActivity implements
-    ZimHostCallbacks, ZimHostContract.View {
+  ZimHostCallbacks, ZimHostContract.View {
 
+  public static final String SELECTED_ZIM_PATHS_KEY = "selected_zim_paths";
+  private static final String TAG = "ZimHostActivity";
+  private static final String IP_STATE_KEY = "ip_state_key";
   @BindView(R2.id.startServerButton)
   Button startServerButton;
   @BindView(R2.id.server_textView)
   TextView serverTextView;
   @BindView(R2.id.recycler_view_zim_host)
   RecyclerView recyclerViewZimHost;
-
   @Inject
   ZimHostContract.Presenter presenter;
-
   @Inject
   AlertDialogShower alertDialogShower;
-
-  private static final String TAG = "ZimHostActivity";
-  private static final String IP_STATE_KEY = "ip_state_key";
-  public static final String SELECTED_ZIM_PATHS_KEY = "selected_zim_paths";
-
   private BooksOnDiskAdapter booksAdapter;
   private BookOnDiskDelegate.BookDelegate bookDelegate;
   private HotspotService hotspotService;
   private String ip;
   private ServiceConnection serviceConnection;
   private ProgressDialog progressDialog;
+  private HashSet<String> selectedBooksId = new HashSet<>();
```

Comment:
this field does not need to exist

Reply:
Then I have to read from the shared preferences every time the user clicks on an item.

Reply:
you only have to save the current set of selectedIds, which you already do.
Had you preserved the original structure where a selection created the list this would have been easier and even easier again in kotlin to the save the list
```
saveToPreference(newList.filter(::isSelected).map(::id))
```